### ISSUE TITLE
Force quit on startup -- win 7/10, amd64, v1.1.0.227

### ISSUE THREAD

**Issue by zp2133** [canton7/SyncTrayzor#193]
Hey all,
Updated to the latest version this morning on two machines, a Win 7 (64 bit) desktop, and Win 10 (64 bit) laptop. Both force quit on startup. Here's the popup error on the Win 10 laptop:

```
>Version: 1.1.0.227; Variant: Installed; Arch: Amd64

> Path: C:\Program Files\SyncTrayzor\SyncTrayzor.exe
>System.UnauthorizedAccessException: Access is denied.

>   at Pri.LongPath.Directory.BeginFind(String normalizedPathWithSearchPattern, WIN32_FIND_DATA& findData)
>   at Pri.LongPath.Directory.<EnumerateFileSystemIterator>d__0.MoveNext()
>   at SyncTrayzor.Services.Conflicts.ConflictFileManager.FindConflictsImpl(String basePath, SlimObservable`1 subject, CancellationToken cancellationToken) in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:line 137
   at SyncTrayzor.Services.Conflicts.ConflictFileManager.<>c__DisplayClass7_0.<FindConflicts>b__0() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:line 110
>--- End of stack trace from previous location where exception was thrown ---
>   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>   at SyncTrayzor.Utils.SlimObservableExtensions.<SubscribeAsync>d__1`1.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Utils\SlimObservable.cs:line 139
>--- End of stack trace from previous location where exception was thrown ---
>   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>   at SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<ScanFoldersAsync>d__31.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:line 204
>--- End of stack trace from previous location where exception was thrown ---
>   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>   at SyncTrayzor.Services.Conflicts.ConflictFileWatcher.<Reset>d__26.MoveNext() in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileWatcher.cs:line 97
--- End of stack trace from previous location where exception was thrown ---
>   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
>   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
>   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

As a side note, I'm currently syncing three computers, the third is a Linux laptop with Syncthing GTK, but no error pops up on that one. The only folder that syncs between the two Win boxes and NOT the Linux box is a windows-only documents directory called 'WIN_DOCS'.

I can provide the log file,of course, if needed.

ZP

**Comment by canton7** [comment]
Looks like it's the same as #191. 

In about 7 minutes I'll have an intermediate build which should fix the issue here: https://ci.appveyor.com/project/canton7/synctrayzor/build/1.1.0.229/artifacts . Would you mind testing it? Thanks

**Comment by zp2133** [comment]
No problem! I'm at work at the moment and can't actively sync on my home network atm, but I can load it and start it up at the very least.

**Comment by canton7** [comment]
Righty, that's built.

If you were trying to start it at work and it crashed, and it now doesn't crash, that's good enough for me!

**Comment by zp2133** [comment]
Downloading now.

**Comment by zp2133** [comment]
No error on startup. All folders scanned. Conflict resolver pops up...which is cool. Didn't know I had some unresolved conflicts.  ;)

**Comment by canton7** [comment]
Thanks! :smile: 

> Didn't know I had some unresolved conflicts. ;)

Yeah I suspect that's the case for a lot of people!

Would you mind looking in the log file for lines containing Failed to enumerate directories in folder or Failed to enumerate files in folder: I'd like to try and figure out what the underlying cause of this is.

**Comment by zp2133** [comment]
Quite a few instances actually. Without listing the whole error output, the following occurs for the Windows folders, 'My Music', 'My Pictures', 'My Videos', etc.

Here's the rub, there's no folder called "C:\Users\HQ\Documents\My Videos" for example. I have a separate drive (D:) on my windows computers that syncs all these folders. So my videos folders are D:\Videos on windows and ~/Videos on Linux. The only folder where I make this exception is the 'WIN_DOCS' folders which is my C:\Users\HQ\Documents. But this isn't throwing any errors.

I'm here to provide more info if you need it...slow day at work.

>2016-01-11 07:18:58.4736 #2500 [Error] SyncTrayzor.Services.Conflicts.ConflictFileManager: Failed to enumerate directories in folder C:\Users\HQ\Documents\My Videos: Access is denied.
> System.UnauthorizedAccessException Access is denied.
> Pri.LongPath.SafeFindHandle BeginFind(System.String, WIN32_FIND_DATA ByRef)    at Pri.LongPath.Directory.BeginFind(String normalizedPathWithSearchPattern, WIN32_FIND_DATA& findData)
>   at Pri.LongPath.Directory.<EnumerateFileSystemIterator>d__0.MoveNext()
>   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
>   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
>   at SyncTrayzor.Services.FilesystemProvider.GetDirectories(String path, String searchPattern, SearchOption searchOption) in C:\projects\synctrayzor\src\SyncTrayzor\Services\FilesystemProvider.cs:line 83
>   at SyncTrayzor.Services.Conflicts.ConflictFileManager.TryGetDirectories(String path, String searchPattern, SearchOption searchOption) in C:\projects\synctrayzor\src\SyncTrayzor\Services\Conflicts\ConflictFileManager.cs:line 196 System.UnauthorizedAccessException: Access is denied.

**Comment by canton7** [comment]
Fixed in 1.1.1
